File: dohiammk_sellgoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;
import entity.Result;

import java.util.List;
import java.util.Map;

public interface BrandService {
    List<TbBrand> findAll();

    /**
     * 当前页数pageNum
     * 每页显示条数pageSize
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findByPage(int pageNum,int pageSize);
    void addBrand(TbBrand brand);
    TbBrand findById(long id);
    void updateBrand(TbBrand brand);
    void deleteBrand(Long[] ids);
    PageResult findSearchPage(TbBrand brand,int pageNum, int pageSize);
    List<Map> selectOptionList();
}

File: dohiammk_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };
    this.findByPage = function (page, size) {
        return $http.get('../brand/findByPage.do?pageNum=' + page + '&pageSize=' + size);
    };
    this.addBrand = function (entity) {
        return $http.post('../brand/addBrand.do', entity);
    };
    this.updateBrand = function (entity) {
        return $http.post('../brand/updateBrand.do', entity);
    };
    this.findById = function (id) {
        return $http.get('../brand/findById.do?id=' + id);
    };
    this.delBrandById = function (ids) {
        return $http.get('../brand/deleteBrand.do?ids=' + ids);
    };
    this.search = function (page, size, searchEntity) {
        return $http.post('../brand/findSearchPage.do?pageNum=' + page + '&pageSize=' + size, searchEntity);
    };
    this.selectOptionList = function () {
        return $http.get('../brand/selectOptionList.do');
    }
});